benchmark.cc

```diff
@@ -110,78 +123,88 @@ graphStat testBuildTime(const uint64_t mapSize, MAP_TYPE mapType, double loadFac
     }
     time2 = std::chrono::high_resolution_clock::now();
     durMs = std::chrono::duration_cast<std::chrono::milliseconds>(time2-time1);
-    if(verbose >= 1) {
+    if(verbose_ >= 1) {
         std::cout << "Time to get every item in " << graphStatistics.getType() << ": " << durMs.count()
                   << " milliseconds\n" << std::endl;
     }
     graphStatistics.msToRead = durMs.count();
+    graphStatistics.finalLoadFactor = map->getLoadFactor();
 
-    delete map;
-    return graphStatistics;
-}
+    if(verbose_ >= 2){
+        map->display();
+    }
 
-int main(int32_t argc, char** argv){
-    read_args(argc, argv);
-    //also compare against unordered map
+    if(verbose_ >= 1){
+        map->stats();
+    }
 
-    graphStat stat = testBuildTime(mapSize, mapType, load);
-    std::cout << stat.getType() << "," << std::to_string(stat.numberOfItems) << "," << std::to_string(stat.loadFactor) << "," << std::to_string(stat.msToFill) << "," << std::to_string(stat.msToRead) << ",\n";
+    delete map;
+    return graphStatistics;
 }
 
 void read_args(int32_t argc, char** argv){
     int32_t opt;
     //why is there no colon between h and v?
-    while((opt = getopt(argc, argv, ":m:s:l:hv")) != -1)
+    while((opt = getopt(argc, argv, ":m:s:l:v:h")) != -1)
     {
         switch(opt)
         {
             case 'v':
-                verbose = 1;
+                //try-catch?
+                verbose_ = atoi(optarg);
                 break;
             case 'h':
                 std::cout << "Usage: test [arguments]\n";
                 std::cout << "\t-m:\tmapType: std (default), stdunordered, chained, or open\n";
                 std::cout << "\t-s:\tmapSize: any uint64_t (default 100,000)\n";
-                std::cout << "\t-l:\tLoadFactor: any double (default 0.7) (0..1 for OPENMAP) (discarded for STDMAP and STDUNORDEREDMAP)\n";
-                std::cout << "\t-v:\tverbose: 1 to enable (default 0)\n";
+                std::cout << "\t-l:\tmaxLoadFactor: any double (default 0.7) (0..1 for OPENMAP) (discarded for STDMAP and STDUNORDEREDMAP)\n";
+                std::cout << "\t-v:\tverbose: default 0, 1 to enable, 2 to enable and print graph\n";
```

Comment:
Great help message.

C++ now has multi-line strings.  The syntax is a little wonky.  This is a very readable alternative.

```
const std::string text =
  "This text is pretty long, but will be "
  "concatenated into just a single string. "
  "The disadvantage is that you have to quote ";
```